### PR TITLE
HPCC-21492 Report an error if embed : activity has an unsupported return type

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
@ghalliday Please can your review

File: ecl/hql/hqlgram2.cpp [hunk 1]
```diff
@@ -1013,7 +1013,11 @@ IHqlExpression * HqlGram::processEmbedBody(const attribute & errpos, IHqlExpress
         }
     }
     else
+    {
+        if (attribs && attribs->queryName()==activityAtom)
```
Comment (ghalliday) [inline ecl/hql/hqlgram2.cpp]: This test is wrong - attribs is a comma list - should be using queryAttributeInList

File: ecl/hql/hqlgram2.cpp [hunk 2]
```diff
@@ -1013,7 +1013,11 @@ IHqlExpression * HqlGram::processEmbedBody(const attribute & errpos, IHqlExpress
         }
     }
     else
```
Comment (ghalliday) [inline ecl/hql/hqlgram2.cpp]: This test condition is wrong.  Embed, activity
- does not support transforms, rows
- does support actions.

**Review by ghalliday — COMMENTED** [review]
@shamser needs more work.

Please can you also add test cases - (in ecl/regress) including where it has more than one attribute.

**Comment by shamser** [comment]
@ghalliday  Please can you review (test case added)

File: ecl/hql/hqlgram2.cpp [hunk 3]
```diff
@@ -918,6 +918,12 @@ IHqlExpression * HqlGram::processEmbedBody(const attribute & errpos, IHqlExpress
     if (!type)
         type.setown(makeVoidType());
 
+    if (type->getTypeCode()!=type_table && type->getTypeCode()!=type_groupedtable && type->getTypeCode()!=type_void)
+    {
+        if (attribs && queryAttributeInList(activityAtom,attribs))
+             reportError(ERR_EMBEDERROR, errpos, "Embedded activities only supports dataset return type or actions");
```
Comment (ghalliday) [inline ecl/hql/hqlgram2.cpp]: picky - 
Embedded activities only support dataset..
or
Embedded activity only supports da

**Pull request merged**